tests/integration/single_table/test_base.py

```diff
@@ -653,18 +653,24 @@ def test_metadata_updated_warning_detect(mock__fit):
         "The 'SingleTableMetadata' is deprecated. "
         "Please use the new 'Metadata' class for synthesizers."
     )
+    expected_input_metadata_updated_message = (
+        'Your metadata has been modified. Metadata modifications cannot be applied to an '
+        'existing synthesizer. Please create a new synthesizer with the modified metadata.'
+    )
 
     # Run
     with warnings.catch_warnings(record=True) as record:
         instance = BaseSingleTableSynthesizer(metadata)
         instance.fit(data)
 
     # Assert
-    assert len(record) == 2
+    assert len(record) == 3
```

Comment:
You shouldn't update this, the warning shouldn't be surfacing here. Moving the position of the `original_metadata` attribute to line 162 as I described above should solve this.

Reply:
This doesn't have to do with the `original_metadata` but the `_input_metadata` which is the object that the user has currently in the running environment. I do see however that we should probably set the flag to `False` once we created the instance of this synthesizer.

Added it here: https://github.com/sdv-dev/SDV/pull/2503/commits/f8b3ef7708ac3500ee6b4dfaa1f8ceff2cc34c20

Reply:
Yes, this is what I meant.